src/internal/github/git.go

```diff
@@ -0,0 +1,51 @@
+package github
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"os/exec"
+	"strings"
+)
+
+// TODO logs
+// TODO Dependency Inversion for tests?
+
+func GetTags(repositoryUrl string) ([]string, error) {
+	log.Printf("Getting tags for repository %s", repositoryUrl)
+
+	var buf bytes.Buffer
+	var bufErr bytes.Buffer
+	cmd := exec.Command("git", "ls-remote", "--tags", "--refs", repositoryUrl)
+	cmd.Stdout = &buf
+	cmd.Stderr = &bufErr
+
+	if err := cmd.Run(); err != nil {
+		log.Printf("Could not get tags for repository %s: %s", repositoryUrl, bufErr.String())
+		return nil, err
+	}
+
+	tags := make([]string, 0)
+	for _, line := range strings.Split(buf.String(), "\n") {
+		if !strings.Contains(line, "refs/tags/") {
+			continue
+		}
+
+		fields := strings.Fields(line)
+		if len(fields) != 2 {
+			return nil, fmt.Errorf("could not parse tags: tags are in wrong format")
+		}
+
+		ref := fields[1]
+		if !strings.HasPrefix(ref, "refs/tags/") {
+			continue
+		}
```

Comment:
**(suggestion)** I guess the first validation was based on the assumption that the git output may change. Following that logic, I'd suggest:

```suggestion
		fields := strings.Fields(line)
		if len(fields) != 2 {
			return nil, fmt.Errorf("could not parse tags: tags are in wrong format")
		}

		ref := fields[1]
		if !strings.HasPrefix(ref, "refs/tags/") {
			return nil, fmt.Errorf("could not parse tags: tags are in wrong format")
		}
```

AFAIK `refs/tags` shall always prefix the version `number` when git is running with the arguments `"--tags", "--refs"`, or am I missing something?

Reply:
That is correct. That is leftover from my previous code that did not use the `--refs` tag (though I think it's technically not possible even if you only use the `--tags` flag)

Reply:
Done in 28f3857e1a844832b815a276a3ec7a4392445429